### PR TITLE
Labeled multiscales

### PR DESCRIPTION
**Pull request by joshmoore**
Captures the changes to the spec for turning labeled images into multiscale arrays. Depending on the decisions around color etc. (see https://github.com/ome/omero-ms-zarr/issues/62#issuecomment-676062914) this may include a top-level "label" key.

see https://github.com/ome/ome-zarr-py/pull/45

### CODE REVIEW

File: spec.md
Comment (sbesson) [inline spec.md:95]: :+1: for adding a version attribute. Is there a case for adding it to each spec? I assume this opens the door for incrementing the version of a given spec independently of the rest.
Probably in v0.2 we might want to make this key mandatory

File: spec.md
Comment (sbesson) [inline spec.md:137]: Assuming this will need to be rationalized with `color` below. Worth a placeholder mentioning it?

File: spec.md
Comment (sbesson) [inline spec.md:157]: Minor RFE, given the removal of the second paragraph, `either` is unnecessary.

File: spec.md
Comment (sbesson) [inline spec.md:50]: Does this mean the `"multiscale"` specification is expected here?

**Review by sbesson — COMMENTED** [review]
Overall, the changes are sensible, in agreement with the discussion of the last few weeks. A few inline comments. Happy to discuss them and convert them as issues/follow-up as necessary.

While reviewing this in relationship with https://github.com/ome/ome-zarr-py/pull/45, one additional confusion is that this document defines the overall layout as `specification` and the individual schemas for the `.zattrs` keys as `metadata` while https://github.com/ome/ome-zarr-py/pull/45 introduces the concept of `Spec` for the latter.

File: spec.md
Comment (joshmoore) [reply spec.md:137]: All in the next commits that I'll open once this is merged.

File: spec.md
Comment (joshmoore) [reply spec.md:95]: A general discussion on how we want to handle the versions (1 vs many; required vs optional) is definitely coming.

File: spec.md
Comment (joshmoore) [reply spec.md:50]: Yes. This has been updated in the color commits, so we should review whether or not it's clear there.

File: spec.md
Comment (joshmoore) [reply spec.md:157]: Also gone in the next set of commits.

File: spec.md
Comment (sbesson) [reply spec.md:137]: :+1:

File: spec.md
Comment (sbesson) [reply spec.md:50]: :+1: looking forward to reviewing these

**Review by sbesson — APPROVED** [review]
LGTM

**Pull request merged**
